src/bin/sol-fbp-runner/Widget.js

```diff
@@ -0,0 +1,222 @@
+(function(window) {
+
+	'use strict'
+
+	function Widget(payload, key) {
+
+		//caching a secure reference to the instance
+		var instance = this;
+		//var selectedPort;
+
+		Widget.prototype.portsCount = 0;
+
+		//unique id for the widget and other unique properties
+		this.raw = JSON.stringify(payload, null, 4);
+		this.key = key;
+		this.widgetName = payload.id; //change to 'name ?'
+		this.uid = payload.path[payload.path.length-1];
+		this.type = payload.type;
+		this.category = payload.category;
+		this.description = payload.description;
+		this.url = payload.url;
+		this.portsIn = [];
+		this.portsOut = [];
+		this.autoSelected = false;
+		this.hidden = true;
+		this.selectedPort = null;
+		this.totalPorts = 0;
+
+		//TODO: we should change the id key to name
+		this.name = "widget" + payload.id;
+
+		//title and subtitle of the widget
+		this.title = $("<span/>").addClass("widget-title").append(this.widgetName);
+		this.subtitle = $("<span/>").addClass("widget-subtitle").append(this.type);
+
+		//info button
+		var info = $("<img/>").attr("src","images/ico_info.png").addClass("info");
+		info.click(function(event) {
+			window.modal.show(instance.raw, instance.type);
+		});
+
+		//creating DOM elements
+		this.displayValue = $("<span/>").addClass("display-value");
+		this.el = $("<div/>").addClass("widget").attr("id",this.uid).appendTo("#container");
+		$("<div/>").addClass("widget-header").appendTo(this.el).append(this.title).append(this.subtitle).append(info);
+		$("<div/>").addClass("widget-display").appendTo(this.el).addClass("widget-data").append(this.displayValue);
+		this.widgetPorts = $("<div/>").addClass("widget-ports").appendTo(this.el);
+
+		//putting the ports in the correct container
+		this.lports = $("<div/>").addClass("port-container").appendTo("#"+this.uid + " .widget-ports");
+		this.rports = $("<div/>").addClass("port-container").appendTo("#"+this.uid + " .widget-ports");
+
+		//reading all in ports
+		var port;
+		payload.ports_in.forEach(function(element, index) {
```

Comment:
this will be empty as when we start, there are no connections yet. Or is this deferred?  If so, until when? In theory connections and disconnections could happen all the time (dynamic), but in practice it's done once at startup (static)